src/main/java/org/opensrp/web/rest/v2/SettingResource.java

```diff
@@ -0,0 +1,220 @@
+			String locationId = RestUtils.getStringFilter(AllConstants.Event.LOCATION_ID, request);
+			String team = RestUtils.getStringFilter(AllConstants.Event.TEAM, request);
+			String teamId = RestUtils.getStringFilter(AllConstants.Event.TEAM_ID, request);
+			String identifier = RestUtils.getStringFilter(SETTING_IDENTIFIER, request);
+			boolean resolveSettings = RestUtils.getBooleanFilter(AllConstants.Event.RESOLVE_SETTINGS, request);
+			Map<String, TreeNode<String, Location>> treeNodeHashMap = null;
+
+			if (StringUtils.isBlank(team) && StringUtils.isBlank(providerId) && StringUtils.isBlank(locationId)
+					&& StringUtils.isBlank(teamId) && StringUtils.isBlank(team) && StringUtils.isBlank(serverVersion)) {
+				return new ResponseEntity<>("All parameters cannot be null for this endpoint",
+						RestUtils.getJSONUTF8Headers(), HttpStatus.BAD_REQUEST);
+			}
+
+			long lastSyncedServerVersion = 0L;
+			if (StringUtils.isNotBlank(serverVersion)) {
+				lastSyncedServerVersion = Long.parseLong(serverVersion) + 1;
+			}
+
+			SettingSearchBean settingQueryBean = new SettingSearchBean();
+			settingQueryBean.setTeam(team);
+			settingQueryBean.setTeamId(teamId);
+			settingQueryBean.setProviderId(providerId);
+			settingQueryBean.setLocationId(locationId);
+			settingQueryBean.setServerVersion(lastSyncedServerVersion);
+			if (StringUtils.isNotBlank(identifier)) {
+				settingQueryBean.setIdentifier(identifier);
+			}
+			if (StringUtils.isNotBlank(locationId)) {
+				settingQueryBean.setResolveSettings(resolveSettings);
+				treeNodeHashMap = getChildParentLocationTree(locationId);
+			}
+
+			List<SettingConfiguration> settingConfigurations = settingService.findSettings(settingQueryBean,
+					treeNodeHashMap);
+			List<Setting> settingList = extractSettings(settingConfigurations);
+
+			return new ResponseEntity<>(gson.toJson(settingList), RestUtils.getJSONUTF8Headers(), HttpStatus.OK);
+
+		}
+		catch (Exception e) {
+			logger.error(MessageFormat.format("Fetching settings failed with the following error {0}.- ", e));
+			return new ResponseEntity<>(RestUtils.getJSONUTF8Headers(), HttpStatus.INTERNAL_SERVER_ERROR);
+		}
+
+	}
+
+	private List<Setting> extractSettings(List<SettingConfiguration> settingConfigurations) {
+		List<Setting> settingList = new ArrayList<>();
+		if (settingConfigurations != null && settingConfigurations.size() > 0) {
+			for (SettingConfiguration settingConfiguration : settingConfigurations) {
+				settingList.addAll(settingConfiguration.getSettings());
+			}
+		}
+
+		return settingList;
+	}
+
+	/**
+	 * Creates or updates a given settings
+	 *
+	 * @param entity {@link Setting} - the settings object
+	 * @return
+	 */
+	@RequestMapping(method = { RequestMethod.POST, RequestMethod.PUT }, consumes = { MediaType.APPLICATION_JSON_VALUE,
+			MediaType.TEXT_PLAIN_VALUE })
+	public ResponseEntity<String> createOrUpdate(@RequestBody String entity) {
```

Comment:
For Post it would be recommended to use /{id} as the URL

Reply:
I am assuming you meant `PUT`

Reply:
Yes PUT. thanks